singer_sdk/streams/core.py

```diff
@@ -320,8 +320,20 @@ def _write_starting_replication_value(self, context: Optional[dict]) -> None:
             ):
                 value = replication_key_value
 
-            elif "start_date" in self.config:
-                value = self.config["start_date"]
+            # Use start_date if it is more recent than the replication_key state
+            if "start_date" in self.config:
+                if not value:
+                    value = self.config["start_date"]
+                else:
+                    value_datetime = cast(datetime.datetime, pendulum.parse(value))
+                    start_datetime = cast(
+                        datetime.datetime, pendulum.parse(self.config["start_date"])
+                    )
```

Comment:
@ericboucher what would happen here if either of these values isn't date-parseable? e.g.  if they are UNIX timestamps.

Reply:
I re-used what was in `get_starting_timestamp` but I can raise a more explicit error if that's useful. Do you prefer that I:
1. raise a more explicit error
2. handle unix timestamps explicitely